File: java-utils/src/main/java/com/cds/common/metrics/sink/JmxSink.java
package com.cds.common.metrics.sink;

import com.codahale.metrics.JmxReporter;
import com.codahale.metrics.MetricRegistry;

import java.util.Properties;

import javax.annotation.concurrent.ThreadSafe;

/**
 * A sink which listens for new metrics and exposes them as namespaces MBeans.
 */
@ThreadSafe
public final class JmxSink implements Sink {
  private JmxReporter mReporter;

  /**
   * Creates a new {@link JmxSink} with a {@link Properties} and {@link MetricRegistry}.
   *
   * @param properties the properties
   * @param registry the metric registry to register
   */
  public JmxSink(Properties properties, MetricRegistry registry) {
    mReporter = JmxReporter.forRegistry(registry).build();
  }

  @Override
  public void start() {
    mReporter.start();
  }

  @Override
  public void stop() {
    mReporter.stop();
  }

  @Override
  public void report() {
  }
}

File: scala-utils/src/main/scala/com/cds/utils/common/metrics/sink/JmxSink.scala
package com.cds.utils.common.metrics.sink

import java.util.Properties

import com.codahale.metrics.{JmxReporter, MetricRegistry}
import org.apache.spark.SecurityManager

/**
  * Created by chendongsheng5 on 2017/6/14.
  */
class JmxSink(val property: Properties, val registry: MetricRegistry,
              securityMgr: SecurityManager) extends Sink {

  val reporter: JmxReporter = JmxReporter.forRegistry(registry).build()

  override def start() {
    reporter.start()
  }

  override def stop() {
    reporter.stop()
  }

  override def report() { }

}
